lusidtools/cocoon/cocoon_printer.py

```diff
@@ -0,0 +1,159 @@
+
+
+def format_instruments_response(response: dict) -> (
+        pd.DataFrame, pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from instrument requests and returns successful, failed and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    :return: pd.DataFrame: Failed responses that LUSID rejected
+    """
+    file_type = "instruments"
+    check_errors_and_success(response[file_type], f"Response from {file_type} request", ["errors", "success"])
+
+    items_success, items_failed, items_error = [], [], []
+
+    # get success and failures
+    items_success = [batch.values[key].name for batch in response[file_type]["success"] for key in batch.values.keys()]
+
+    if check_for_failed_items(response[file_type]["success"]):
+        items_failed = [batch.failed[key].name for batch in response[file_type]["success"] for key in
+                        batch.failed.keys()]
+
+    errors = get_errors_from_response(response[file_type]["errors"])
+
+    return (pd.DataFrame(items_success, columns=["successful items"]),
+            errors,
+            pd.DataFrame(items_failed, columns=["failed_items"]))
+
+
+def format_portfolios_response(response: dict) -> (pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from portfolio requests and returns successful and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    """
+    file_type = "portfolios"
+    check_errors_and_success(response[file_type], f"Response from {file_type} request", ["errors", "success"])
+
+    # get success
+    items_success = [batch.id.code for batch in response[file_type]["success"]]
+
+    errors = get_errors_from_response(response[file_type]["errors"])
+
+    return (pd.DataFrame(items_success, columns=["successful items"]),
+            errors)
+
+
+def format_transactions_response(response: dict, file_type: str = "transactions") -> (pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from transaction requests and returns successful and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :param str file_type: file_type of Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    """
+
+    check_errors_and_success(response[file_type], f"Response from {file_type} request", ["errors", "success"])
+
+    # get success
+    items_success = [batch.href for batch in response[file_type]["success"]]
```

Comment:
Do you want to return the href here? Or should you split it out into something easier to read?